querycheck/expect_resource_known_values.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package querycheck
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-testing/querycheck/queryfilter"
+	"github.com/hashicorp/terraform-plugin-testing/tfjsonpath"
+)
+
+var _ QueryResultCheck = expectResourceKnownValues{}
+var _ QueryResultCheckWithFilters = expectResourceKnownValues{}
+
+type expectResourceKnownValues struct {
+	listResourceAddress string
+	filter              queryfilter.QueryFilter
+	knownValues         []KnownValueCheck
```

Comment:
super nit 🦸🏻 - These are checks, not known values 🙂  , I think `knownValueChecks` would be a little easier to read further down in the code.